contracts/token/StakableToken.sol

```diff
@@ -2,17 +2,21 @@ pragma solidity ^0.4.11;
 
 import "installed_contracts/zeppelin/contracts/token/StandardToken.sol";
 
+contract RegistryInterface {
+  function isOracleRegistered(address _key) returns (bool);
+  function isNodeRegistered(address _key) returns (bool);
+}
+
 contract StakableToken is StandardToken {
 
   // Mapping of address staking => staked amount
   mapping (address => uint256) public stakedAmount;
-  // Mapping of address staking => count of transactions in current throttle epoch
-  mapping (address => uint256) public epochTransactionCount;
+  mapping (address => uint256) public nodeStakedAmount;
+  mapping (address => uint256) public oracleStakedAmount;
+  mapping (address => uint256) public investorStakedAmount;
```

Comment:
We shouldn't do it this way, cause you're missing on behalf of whom you're staking.

Reply:
Good point, I can make it a mapping of mappings